WooCommerce/Classes/Model/Coupon+Woo.swift

```diff
@@ -0,0 +1,84 @@
+import Foundation
+import Yosemite
+
+extension Coupon.DiscountType {
+    /// Localized name to be displayed for the discount type.
+    ///
+    var localizedName: String {
+        switch self {
+        case .percent:
+            return Localization.percentageDiscount
+        case .fixedCart:
+            return Localization.fixedCartDiscount
+        case .fixedProduct:
+            return Localization.fixedProductDiscount
+        case .other:
+            return Localization.otherDiscount
+        }
+    }
+
+    private enum Localization {
+        static let percentageDiscount = NSLocalizedString("Percentage Discount", comment: "Name of percentage discount type")
+        static let fixedCartDiscount = NSLocalizedString("Fixed Cart Discount", comment: "Name of fixed cart discount type")
+        static let fixedProductDiscount = NSLocalizedString("Fixed Product Discount", comment: "Name of fixed product discount type")
+        static let otherDiscount = NSLocalizedString("Other", comment: "Generic name of non-default discount types")
+    }
+}
+
+// MARK: - Coupon expiry status
+//
+extension Coupon {
+    /// Expiry status for Coupons.
+    ///
+    var expiryStatus: ExpiryStatus {
+        guard let expiryDate = dateExpires else {
+            return .active
+        }
+
+        guard let gmtTimeZone = TimeZone(identifier: "GMT") else {
+            return .expired
+        }
+        let now = Date()
+
+        var calender = Calendar.current
+        calender.timeZone = gmtTimeZone
+
+        let result = calender.compare(expiryDate, to: now, toGranularity: .day)
+        if result == .orderedDescending {
+            return .active
+        }
+        return .expired
```

Comment:
nit: can be slightly simplified to one line
```suggestion
        return result == .orderedDescending ? .active: .expired
```

Reply:
I've updated this in fd83ce3ae712df2c3c0f1ac2cfc9c72751bad772 - thank you!